src/rgb_xyb.rs

```diff
@@ -39,66 +39,70 @@ const NEG_OPSIN_ABSORBANCE_BIAS: [f32; 3] = [-K_B0, -K_B1, -K_B2];
-pub fn xyb_to_linear_rgb(input: &[[f32; 3]]) -> Vec<[f32; 3]> {
+pub fn xyb_to_linear_rgb(mut input: Vec<[f32; 3]>) -> Vec<[f32; 3]> {
     let mut biases_cbrt = NEG_OPSIN_ABSORBANCE_BIAS;
     for bias in &mut biases_cbrt {
         *bias = cbrtf(*bias);
     }
 
-    input
-        .iter()
-        .map(|pix| {
-            // Color space: XYB -> RGB
-            let mut gamma_rgb = [pix[1] + pix[0], pix[1] - pix[0], pix[2]];
-            for ((rgb, bias_cbrt), neg_bias) in gamma_rgb
-                .iter_mut()
-                .zip(biases_cbrt.iter())
-                .zip(NEG_OPSIN_ABSORBANCE_BIAS.iter())
-            {
-                *rgb -= *bias_cbrt;
-                // Undo gamma compression: linear = gamma^3 for efficiency.
-                let tmp = (*rgb) * (*rgb);
-                *rgb = tmp.mul_add(*rgb, *neg_bias);
-            }
+    for pix in &mut input {
+        // Color space: XYB -> RGB
+        let mut gamma_rgb = [pix[1] + pix[0], pix[1] - pix[0], pix[2]];
+        for ((rgb, bias_cbrt), neg_bias) in gamma_rgb
+            .iter_mut()
+            .zip(biases_cbrt.iter())
+            .zip(NEG_OPSIN_ABSORBANCE_BIAS.iter())
+        {
+            *rgb -= *bias_cbrt;
+            // Undo gamma compression: linear = gamma^3 for efficiency.
+            let tmp = (*rgb) * (*rgb);
+            *rgb = tmp.mul_add(*rgb, *neg_bias);
+        }
+
+        let mut out = gamma_rgb;
+        out[0] = INVERSE_OPSIN_ABSORBANCE_MATRIX[0] * gamma_rgb[0];
+        out[1] = INVERSE_OPSIN_ABSORBANCE_MATRIX[3] * gamma_rgb[0];
+        out[2] = INVERSE_OPSIN_ABSORBANCE_MATRIX[6] * gamma_rgb[0];
+        out[0] = INVERSE_OPSIN_ABSORBANCE_MATRIX[1].mul_add(gamma_rgb[1], out[0]);
+        out[1] = INVERSE_OPSIN_ABSORBANCE_MATRIX[4].mul_add(gamma_rgb[1], out[1]);
+        out[2] = INVERSE_OPSIN_ABSORBANCE_MATRIX[7].mul_add(gamma_rgb[1], out[2]);
+        out[0] = INVERSE_OPSIN_ABSORBANCE_MATRIX[2].mul_add(gamma_rgb[2], out[0]);
+        out[1] = INVERSE_OPSIN_ABSORBANCE_MATRIX[5].mul_add(gamma_rgb[2], out[1]);
+        out[2] = INVERSE_OPSIN_ABSORBANCE_MATRIX[8].mul_add(gamma_rgb[2], out[2]);
 
-            let mut out = gamma_rgb;
-            out[0] = INVERSE_OPSIN_ABSORBANCE_MATRIX[0] * gamma_rgb[0];
-            out[1] = INVERSE_OPSIN_ABSORBANCE_MATRIX[3] * gamma_rgb[0];
-            out[2] = INVERSE_OPSIN_ABSORBANCE_MATRIX[6] * gamma_rgb[0];
-            out[0] = INVERSE_OPSIN_ABSORBANCE_MATRIX[1].mul_add(gamma_rgb[1], out[0]);
-            out[1] = INVERSE_OPSIN_ABSORBANCE_MATRIX[4].mul_add(gamma_rgb[1], out[1]);
-            out[2] = INVERSE_OPSIN_ABSORBANCE_MATRIX[7].mul_add(gamma_rgb[1], out[2]);
-            out[0] = INVERSE_OPSIN_ABSORBANCE_MATRIX[2].mul_add(gamma_rgb[2], out[0]);
-            out[1] = INVERSE_OPSIN_ABSORBANCE_MATRIX[5].mul_add(gamma_rgb[2], out[1]);
-            out[2] = INVERSE_OPSIN_ABSORBANCE_MATRIX[8].mul_add(gamma_rgb[2], out[2]);
-            out
-        })
-        .collect()
+        pix[0] = out[0];
```

Comment:
I think we can also avoid this extra step by writing directly to `pix` instead of having an `out` array